witchcraft/server_routes.go

```diff
@@ -172,3 +179,25 @@ func heap(w http.ResponseWriter, _ *http.Request) {
 		return
 	}
 }
+
+func getSecretRefreshable(ctx context.Context, diagnosticsConfig config.RefreshableDiagnosticsConfig) (refreshable.String, error) {
+	secretFromConfig := diagnosticsConfig.DebugSharedSecret()
+	secretFilePath := diagnosticsConfig.DebugSharedSecretFile()
+	// if both fields are undefined, then the server doesn't use a secret for the /debug/diagnostics/* route
+	if secretFromConfig.CurrentString() == "" && secretFilePath.CurrentString() == "" {
+		return refreshable.NewString(refreshable.NewDefaultRefreshable("")), nil
+	}
+
+	if secretFromConfig.CurrentString() != "" {
+		return secretFromConfig, nil
+	}
+	fileRefreshable, err := refreshablefile.NewFileRefreshable(ctx, secretFilePath.CurrentString())
+	if err != nil {
+		return nil, err
+	}
+	secretStringFromFileRefreshable := refreshable.NewString(fileRefreshable.Map(func(i interface{}) interface{} {
+		secretBytes := i.([]byte)
```

Comment:
can we be extra safe here and do:
```
secretBytes, ok := i.([]byte)
if !ok { return "" }
```
just to avoid any panics? We typically do this in places even where we are confident in the type assertion always succeeding.